src/actions/vstudio/vs2005_csproj.lua

```diff
@@ -368,33 +368,89 @@
+				local packageAPIInfo = vstudio.nuget2010.packageAPIInfo(prj, package)
+
+				local cfg = p.project.getfirstconfig(prj)
+				local action = premake.action.current()
+				local targetFramework = cfg.dotnetframework or action.vstudio.targetFramework
+
+				-- This is a bit janky. To compare versions, we extract all
+				-- numbers from the given string and right-pad the result with
+				-- zeros. Then we can just do a lexicographical compare on the
+				-- resulting strings.
+				--
+				-- This is so that we can compare version strings such as
+				-- "4.6" and "net451" with each other.
+
+				local function makeVersionComparable(a)
+					local numbers = ""
+
+					for number in a:gmatch("%d") do
+						numbers = numbers .. number
+					end
+
+					return string.format("%-10d", numbers):gsub(" ", "0")
+				end
+
+				local targetVersion = makeVersionComparable(targetFramework)
+
+				-- Figure out what folder contains the files for the nearest
+				-- supported .NET Framework version.
+
+				local files = {}
+
+				local bestVersion, bestFolder
+
+				for _, file in ipairs(packageAPIInfo.packageEntries) do
+					-- If this exporter ever supports frameworks such as
+					-- "netstandard1.3", "sl4", "sl5", "uap10", "wp8" or
+					-- "wp71", this code will need changing to match the right
+					-- folders.
+
+					local folder = file:match("^lib\\net(%d+)\\")
+
+					if folder and path.hasextension(file, ".dll") then
+						files[folder] = files[folder] or {}
+						table.insert(files[folder], file)
+
+						local version = makeVersionComparable(file:match("lib\\net(%d+)\\"))
+
+						if version <= targetVersion and (not bestVersion or version > bestVersion) then
+							bestVersion = version
+							bestFolder = folder
+						end
+					end
+				end
+
+				if not bestVersion then
+					p.error("NuGet package '%s' is not compatible with project '%s' .NET Framework version '%s'", id, prj.name, targetFramework)
 				end
 
-				_p(3, '<Private>True</Private>')
-				_p(2, '</Reference>')
+				-- Now, add references for all DLLs in that folder.
+
+				for _, file in ipairs(files[bestFolder]) do
+					-- There's some stuff missing from this include that we
+					-- can't get from the API and would need to download and
+					-- extract the package to figure out. It looks like we can
+					-- just omit it though.
+					--
+					-- So, for example, instead of:
+					--
+					-- <Reference Include="nunit.framework, Version=3.6.1.0,
+					-- <Culture=neutral, PublicKeyToken=2638cd05610744eb,
+					-- <processorArchitecture=MSIL">
+					--
+					-- We're just outputting:
+					--
+					-- <Reference Include="nunit.framework">
+
+					_x(2, '<Reference Include="%s">', path.getbasename(file))
+					_x(3, '<HintPath>%s</HintPath>', vstudio.path(prj, p.filename(prj.solution, string.format("packages\\%s.%s\\%s", id, packageAPIInfo.verbatimVersion or packageAPIInfo.version, file))))
+					_p(3, '<Private>True</Private>')
```

Comment:
One thing to note about this which we've had to fix on our internal version - this doesn't respect NoCopyLocal. On our version we replaced this line with:
local cfg = project.getfirstconfig(prj)
if premake.config.isCopyLocal(prj, package, cfg) then
_p(3, 'True')
else
_p(3, 'False')
end